### PR TITLE
CB-11277 Allow to create new datahub, when the env is starting. If th…

### PR DESCRIPTION
**Pull request by topolyai5**
…e freeipa and datalake started but in dh starting phase, we can allow to create a new datahub. We will check the freeipa state instead of the environment state.

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java [hunk 1]
```diff
@@ -49,11 +53,15 @@ public StackV4Response post(DistroXV1Request request) {
...
         DetailedEnvironmentResponse environment = Optional.ofNullable(environmentClientService.getByName(request.getEnvironmentName()))
                 .orElseThrow(() -> new BadRequestException("No environment name provided hence unable to obtain some important data"));
-        if (environment != null && environment.getEnvironmentStatus() != EnvironmentStatus.AVAILABLE) {
-            throw new BadRequestException(String.format("Environment state is %s instead of AVAILABLE", environment.getEnvironmentStatus()));
+        if (environment == null) {
+            throw new BadRequestException(String.format("Cannot fetch the environment by name: %s", request.getEnvironmentName()));
+        }
+        DescribeFreeIpaResponse freeipa = freeipaClientService.getByEnvironmentCrn(environment.getCrn());
+        if (!freeipa.getStatus().isAvailable()) {
+            throw new BadRequestException(String.format("Freeipa state is %s instead of AVAILABLE", freeipa.getStatus()));
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java:61]: If you want to provision a Data Hub then the FreeIPA instance must be running in the '%s' Environment.

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java [hunk 2]
```diff
@@ -49,11 +53,15 @@ public StackV4Response post(DistroXV1Request request) {
     private void validate(DistroXV1Request request) {
         DetailedEnvironmentResponse environment = Optional.ofNullable(environmentClientService.getByName(request.getEnvironmentName()))
                 .orElseThrow(() -> new BadRequestException("No environment name provided hence unable to obtain some important data"));
-        if (environment != null && environment.getEnvironmentStatus() != EnvironmentStatus.AVAILABLE) {
-            throw new BadRequestException(String.format("Environment state is %s instead of AVAILABLE", environment.getEnvironmentStatus()));
+        if (environment == null) {
+            throw new BadRequestException(String.format("Cannot fetch the environment by name: %s", request.getEnvironmentName()));
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java:57]: '%s' Environment does not exist.

**Review by biharitomi — APPROVED** [review]
Please update the BadREquest messages as @doktoric requested, otherwise LGTM

File: core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java (hunk 2 above)
Comment (topolyai5) [reply core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java:57]: ok, I will do it

**Pull request merged**